android/build.gradle

```diff
@@ -96,6 +98,22 @@ if (androidSdkInstalled) {
         publicApiDocs
     }
 
+    task addVersionToConscryptClass(type: Copy) {
+        ext {
+            parsedVersion = VersionNumber.parse(version)
+        }
+        from "${rootDir}/common/src/main/java/org/conscrypt/Conscrypt.java"
+        into 'build/modified'
+        filter {
+            line -> line.replace("new Version(-1, -1, -1)",
+                    "new Version(${parsedVersion.getMajor()}, ${parsedVersion.getMinor()}, ${parsedVersion.getMicro()})")
```

Comment:
@flooey instead of overwriting Conscrypt.java, could we just add metadata to the jar that we can lookup on classpath?

Reply:
Yep, good idea.  Added a properties file that gets read at runtime.